src/components/dataExport/DataExport.tsx

```diff
@@ -0,0 +1,169 @@
+import { Button, Dropdown, LoadingScreen } from "components/elements/customUI/CustomUI";
+
+
+const DataExport = () => {
+    /* Hooks */
+    const navigate = useNavigate();
+    const fetch = useFetch();
+    const loading = useLoading();
+    const notification = useNotification();
+    const trigger = useTrigger();
+
+    /* Base variables */
+    const [sourceSystemDropdownItems, setSourceSystemDropdownItems] = useState<DropdownItem[]>([]);
+    const initialValues: {
+        sourceSystemID: string | undefined
+    } = {
+        sourceSystemID: undefined
+    };
+
+    /* OnLoad, redirect to home if user is not logged in yet */
+    trigger.SetTrigger(() => {
+        if (!KeycloakService.IsLoggedIn()) {
+            navigate('/');
+        }
+    }, []);
+
+    /* OnLoad, fetch source systems */
+    fetch.Fetch({
+        Method: GetSourceSystems,
+        Handler: (sourceSystems: SourceSystem[]) => {
+            const sourceSystemDropdownItems: DropdownItem[] = [];
+
+            sourceSystems.map(sourceSytem => {
+                sourceSystemDropdownItems.push({
+                    label: sourceSytem['schema:name'] ?? sourceSytem['schema:identifier'],
+                    value: sourceSytem['@id'] ?? sourceSytem['schema:identifier']
+                });
+            });
+
+            setSourceSystemDropdownItems(sourceSystemDropdownItems);
+        }
+    });
+
+    return (
+        <div className="h-100 d-flex flex-column">
+            {/* Render header*/}
+            <Header span={10}
+                offset={1}
+            />
+
+            <Container fluid className="flex-grow-1 overflow-hidden py-5">
+                <Row className="h-100">
+                    <Col lg={{ span: 4, offset: 4 }}
+                        className="h-100 d-flex align-items-center"
+                    >
+                        <Formik initialValues={initialValues}
+                            onSubmit={async (values) => {
+                                await new Promise((resolve) => setTimeout(resolve, 100));
+
+                                /* Start loading */
+                                loading.Start();
+
+                                /* Post data export request */
+                                try {
+                                    await ScheduleDataExport({
+                                        dataExportKey: "$[ods:sourceSystemID]",
+                                        dataExportValue: values.sourceSystemID
+                                    });
+
+                                    notification.Push({
+                                        key: `dataExport_${Math.random()}`,
+                                        message: 'Data Export scheduled successfully, your download link will be sent via email',
+                                        template: 'success'
+                                    });
+                                } catch {
+                                    notification.Push({
```

Comment:
maybe put these push calls into a single function? 
```
const pushNotification = (message: string, template: 'success' | 'error') => {
    notification.Push({
        key: `dataExport_${Math.random()}`,
        message,
        template
    });
};
```

Reply:
The Push function already is the function you are describing here, so that would duplicate it.